libp2p/switch.nim

```diff
@@ -60,7 +60,7 @@ proc secure(s: Switch, conn: Connection): Future[Connection] {.async, gcsafe.} =
   if manager.len == 0:
     raise newException(CatchableError, "Unable to negotiate a secure channel!")
 
-  result = await s.secureManagers[manager].secure(conn)
+  result = await s.secureManagers[manager].secure(conn, outgoing = true)
```

Comment:
no need to pass a flag here, calling `secure` is already enough to signal that we're the initiator of the connection, incoming connections are handled inside the `handle` method in `LPProtocol`.

Reply:
Well, then there is a issue with `testswitch`, without such flags I would not be able to understand which is which , cos both calls are on secure

Reply:
And actually in `switch.nim` both incoming and outgoing are calling secure in the end , that's why I had to add this, am I missing something?

Reply:
maybe just the way the test is done is wrong, could you check that too please?

Reply:
I'm not sure why you need that distinction, but either way, when calling `secure` it means that you're the initiator of the connection, if the request arrives in the secure handler, then that is the receiving end.  Basically, when calling `secure` just invoke the `handshake` with the initiator flag set to true, in `handler`, call it with the flag set to false.

Reply:
here outgoing:
https://github.com/status-im/nim-libp2p/blob/master/libp2p/switch.nim#L168-L178
here incoming:
https://github.com/status-im/nim-libp2p/blob/master/libp2p/switch.nim#L183-L193 ->
https://github.com/status-im/nim-libp2p/blob/master/libp2p/switch.nim#L53-L64

Both call the same secure.

Reason is noise handshake is different between initiator/not initiator

Reply:
It still should be the same, calling secure would mean you're the initiator. I'm not 100% it's being called for the incoming connection, I think there is a reason but I need to remember exactly what's the use case.

Reply:
@dryajov I'm interested in fixing this btw, I dunno I used the switch test and that one definitely is using on both ends the same secure, it might be that the test is not correctly written?